wa/workloads/antutu/__init__.py

```diff
@@ -109,16 +145,98 @@ def extract_scores(self, context, regex_version):
 
+        context.add_metric('CPU Total Score', cpu_result, lower_is_better=False)
+        context.add_metric('GPU Total Score', gpu_result, lower_is_better=False)
+        context.add_metric('UX Total Score', ux_result, lower_is_better=False)
+        context.add_metric('MEM Total Score', mem_result, lower_is_better=False)
+
+        #Calculate overall scores
+        overall_result = float(cpu_result + gpu_result + ux_result + mem_result)
+        context.add_metric('Overall Score', overall_result, lower_is_better=False)
+
     def update_output(self, context):
         super(Antutu, self).update_output(context)
+        if self.version.startswith('10'):
+            self.extract_scores(context, self.regex_matches_v10)
         if self.version.startswith('9'):
             self.extract_scores(context, self.regex_matches_v9)
         if self.version.startswith('8'):
             self.extract_scores(context, self.regex_matches_v8)
         if self.version.startswith('7'):
             self.extract_scores(context, self.regex_matches_v7)
+
+
+class AntutuBDP(Workload):
+
+    name = "antutu_bdp"
+    description = '''
+    Workload for executing the BDP versions of the Antutu APK.
+
+    This will only work with specific APKS provided by Antutu but does
+    unlock command line automation and the capturing of a result file
+    as opposed to using UiAuto and Regex.
+
+    Known working version: 10.4.3-domesticAndroidFullBdp
+    '''
+    package_name = 'com.antutu.ABenchMark'
+    test_dir = '/sdcard/Android/data/com.antutu/ABenchMark/files/.antutu/'
+    result_file = '/sdcard/Documents/antutu/last_result.json'
+
+    def initialize(self, context):
+        super(AntutuBDP, self).initialize(context)
+        #Install the supporting benchmark
+        supporting_apk = context.get_resource(ApkFile(self, package='com.antutu.benchmark.full'))
+        self.target.install(supporting_apk)
+        #Launch the apk to initialize the test dir, then kill it
+        self.target.execute('am start {}/com.android.module.app.ui.test.activity.ActivityScoreBench'.format(self.package_name))
+        self.target.execute('am force-stop {}'.format(self.package_name))
+        #Copy the settings.xml to the test dir
+        settings_xml = context.get_resource(File(self, 'settings.xml'))
+        self.target.push(settings_xml, self.test_dir)
+        #Ensure the orientation is set to portrait
+        self.target.set_rotation(0)
+        #Remove any pre-existing test results
+        self.target.execute('rm {}'.format(self.result_file))
+
+    def run(self, context):
+        super(AntutuBDP, self).run(context)
+        #Launch the tests
+        activity = 'com.android.module.app.ui.start.ABenchMarkStart --ez isExternal true --es whereTo "test"'
+        self.target.execute('am start -n {}/{}'.format(self.package_name, activity))
+        #Wait 10 minutres, then begin polling every 30s for the test result to appear
```

Comment:
Question: Is it possible for these tests to run quicker than this or is it expected / required that these tests take at least this length of time to run?

Reply:
The tests are currently running for approximately 18 minutes on the latest flagships. I figured 10 minutes gives enough room for improvement for the foreseeable but I could parameterise this if preferred? (I will fix that typo though in the comment)

Reply:
Fair enough, that sounds reasonable for now.